### DISCUSSION TITLE
Data corruption in Embedded Server

### DISCUSSION THREAD

**Discussion by yjagota** [ravendb/ravendb#13350] [Q&A]
Hello,

In a console application with Embedded RavenDb, if the application is closed immediately after doing a db operation, a lot of times the data gets corrupted with the exception: "VoronUnrecoverableErrorException: Error syncing the data file. The last sync tx is ...".

Is there a way to ask the embedded server to shutdown or commit all changes to the disk like a flush or checkpoint?

**Comment by ayende** [comment]
That is really surprising. Can you show us how you reproduce this?

**Reply by yjagota** [reply]
I primarily faced this in a self-updating service that needs an external app to update it. The external app issues a ServiceController.Stop and if stop is successful, it updates the service. But many a times, the next time the service (which runs an embedded server) is started, the error is thrown.

So, I tried this with a console app. I could reproduce it but it is very rare. If the app is started, does some heavy data op, and then closes almost as soon as the operation is done (using Environment.Exit or an external app closes it) the above error is encountered.

**Reply by ayende** [reply]
Can you send us the console app in question?
